packages/hoppscotch-cli/src/utils/request.ts

```diff
@@ -0,0 +1,336 @@
+      const { config } = baseResponse;
+      const runnerResponse: RunnerResponseInfo = {
+        ...baseResponse,
+        path: x.path,
+        endpoint: GRequest.endpoint(config.url),
+        method: GRequest.method(config.method),
+        body: baseResponse.data,
+      };
+
+      // !NOTE: Temporary `config.supported` check
+      if ((config as RequestConfig).supported === false) {
+        status = 501;
+        runnerResponse.status = status;
+        runnerResponse.statusText = responseErrors[status];
+      }
+
+      return runnerResponse;
+    } catch (err) {
+      let status: number;
+      let statusText: string;
+      const runnerResponse: RunnerResponseInfo = {
+        path: x.path,
+        endpoint: "",
+        method: "GET",
+        body: {},
+        statusText: "",
+        status: 0,
+        headers: [],
+      };
+
+      if (axios.isAxiosError(err)) {
+        runnerResponse.method = GRequest.method(err.config.method);
+        runnerResponse.endpoint = GRequest.endpoint(err.config.url);
+
+        // !NOTE: Temporary `config.supported` check
+        if ((err.config as RequestConfig).supported === false) {
+          status = 501;
+          statusText = responseErrors[status];
+        } else if (!err.response) {
+          status = 408;
+          statusText = responseErrors[status];
+        } else {
+          status = err.response.status;
+          statusText = err.response.statusText;
+        }
+      } else {
+        status = 600;
+        statusText = responseErrors[status];
+      }
+      runnerResponse.status = status;
+      runnerResponse.statusText = statusText;
+
+      return runnerResponse;
+    }
+  };
+
+/**
+ * The request parser from the collection JSON.
+ * @param collection The collection object parsed from the JSON.
+ * @param rootPath The folder path.
+ * @returns RequestStack[] - Created request-stacks successfully parsed
+ * from HoppCollection.
+ * HoppCLIError - On error while parsing HoppCollection.
+ */
+export const requestsParser = (
+  collection: HoppCollection<HoppRESTRequest>,
+  rootPath: string = "$ROOT"
+): TE.TaskEither<HoppCLIError, RequestStack[]> =>
+  pipe(
+    /**
+     * Mapping collection's HoppRESTRequests to EffectiveHoppRESTRequest,
+     * then running preRequestScriptRunner over them and mapping output
+     * to generate RequestStack[] which is binded to PARSED_REQUESTS.
+     */
+    collection.requests,
+    A.map(
+      (hoppRequest) =>
+        Object({
+          ...hoppRequest,
+          effectiveFinalBody: null,
+          effectiveFinalHeaders: [],
+          effectiveFinalParams: [],
+          effectiveFinalURL: S.empty,
+        }) as EffectiveHoppRESTRequest
+    ),
+    A.map(preRequestScriptRunner),
+    TE.sequenceArray,
+    TE.map(RA.toArray),
+    TE.map(
+      A.map((effHoppRequest) =>
+        createRequest(`${rootPath}/${collection.name}`, effHoppRequest)
+      )
+    ),
+    TE.bindTo("PARSED_REQUESTS"),
```

Comment:
What is the point of binding if you are not aggregating TEs. You are just chaining them in a sequence. A normal chain should suffice right ?

Reply:
Actually, I used bindTo, to help me keep track of output so far, visible to me while working on next part. If It seems ambiguous, I will update 👍🏽

Reply:
Umm, do not use binds for letting you know things, that is what comments are for 🤣
You can place a comment near the chain/map to explain what and why.